src/main/kotlin/org/rust/lang/core/dfa/CFGBuilder.kt

```diff
@@ -216,8 +217,15 @@ class CFGBuilder(
     override fun visitLabelDecl(labelDecl: RsLabelDecl) = finishWith(pred)
 
     override fun visitExprStmt(exprStmt: RsExprStmt) {
-        val exprExit = process(exprStmt.expr, pred)
-        finishWithAstNode(exprStmt, exprExit)
+        val expr = exprStmt.expr
+        // Conditionally disabled code should be ignored since it does not affect the execution
+        // https://doc.rust-lang.org/reference/expressions.html#expression-attributes
+        if (expr is RsOuterAttributeOwner && !expr.isEnabledByCfgSelf) {
```

Comment:
Looks like some our classes, doesn't implement `RsOuterAttributeOwner`, though actually they should, e.g. `RsMacroExpr` and `RsCallExpr`. Because of this we don't have unused variable warning in code like this:
```rust
fn main() {
    let a/*caret*/ = 1;
    #[cfg(windows)] println!("{}", a);
}
```

```rust
fn main() {
    let a/*caret*/ = 1;
    #[cfg(windows)] foo(a);
}

fn foo(a: i32) {}
```

Reply:
Thank you for finding these cases!

`RsMacroExpr` is not `RsOuterAttributeOwner` intentionally, see https://github.com/intellij-rust/intellij-rust/commit/0eee09bec8052865d1836a994388a03857bd6d14 (but I don't know the exact reason actually). However there is an easy workaround: check `cfg` attributes right inside `visitMacroExpr`.

But as for `RsCallExpr`, there is a real problem with the parser, so `#[cfg(x)] foo(a)` is parsed as `{#[cfg(x)] foo]} {(a)}`: 
<img width="1021" alt="Screenshot 2021-02-03 at 19 05 00" src="https://user-images.githubusercontent.com/4854600/106774038-c4cc3200-6652-11eb-986a-1d44a0a6f5d0.png">
I've opened a separate issue for that https://github.com/intellij-rust/intellij-rust/issues/6797